[PE-1704] Refine golangci lint execution

kevinkirkup opened the pull request:
**Jira Issue**: [PE-1704]

```release-note:
Enable all tfproviderlint and golangci-lint checks.
```

## Details
Enable linters and all check as well as fix any issues.
Make sure we run these for each CI/CD build.
<!--- Add any detailed information here -->


[PE-1704]: https://pureport.atlassian.net/browse/PE-1704

Try75 commented on Jenkinsfile:
```diff
@@ -16,6 +16,11 @@ pipeline {
     }
+
+    triggers {
+      cron('0 0 * * *')
```
maybe 0 22 * * *? I'm just thinking not at midnight. I couldn't really find a way to see all the scheduled jobs in Jenkins to see when it was least used.

kevinkirkup replied:
How about '1 H * * *' - Sometime between 1am and 2am?

Try75 replied:
"30 1 * * *"?

kevinkirkup replied:
Updated.

kevinkirkup commented:
Also added a fix for the Locations Data Source to add the new virtual pod for Dev1.

Commit:
[PE-1704] Run all linters and fix deprecation warnings

Fix deprecation issues related to partial state handle.
Make sure we run all checks when running linter.

Commit:
[PE-1704] Run linters during CICD builds

Commit:
Need to set golangci-lit cache directory

Commit:
Merge branch 'develop' into feature/PE-1704_Refine_golangci-lint_execution
